syntax = "proto3";

package proto;

message SearchResult {
	repeated EntityContent entity_contents = 1; 
	bytes    debugInfo = 2;
}

message EntityContent {
	         bytes   entity_id = 1; 
	         bytes   index     = 2; // 参与排序的值
	repeated Pair    pairs     = 3;
}

message Pair {
	bytes key = 1;
	bytes value = 2;
}

message Entity {
			 bytes kgc_id 		= 1;
			 bytes entity_id 	= 2;
	         bytes   index      = 3; // 参与排序的值
	repeated Triad triads 		= 4;
}

message Triad {
	bytes predicate 	= 1;
	bytes value 		= 2;
	bytes to_entity_id 	= 3;
	bytes to_kgc_id 	= 4;
	bytes source 		= 5;
	int64 priority 		= 6;
	bytes op	 		= 7;
	int64 op_time		= 8;
}

